cfme/fixtures/vm_console.py

```diff
@@ -0,0 +1,27 @@
+# -*- coding: utf-8 -*-
+"""Fixtures that VMware console tests use to configure VM Console type and start websocket."""
+import pytest
+
+from cfme.utils.log import logger
+
+
+@pytest.fixture(scope="module")
+def configure_vmware_console_for_test(appliance, provider, request):
+    """Configure VMware Console to use console type as required by test(like VNC, VMware WebMKS)."""
+    logger.info("Changing VMware console suppport configuration to {} "
+            .format(request.param['console_type']))
+    appliance.server.settings.update_vmware_console({'console_type': request.param['console_type']})
+
+
+@pytest.yield_fixture(scope="module")
+def configure_websocket(appliance):
+    """Enable websocket role if it is disabled.
+
+    Currently the fixture cfme/fixtures/base.py:27,
+    disables the websocket role to avoid intrusive popups.
+    """
+    appliance.server.settings.enable_server_roles('websocket')
```

Comment:
Is there anyway to make this code smarter to enable it only it hasn't been enabled, and later disable it only it hasn't been disabled?

Reply:
I think that is what we were doing earlier and then we changed it in this PR, don't recall why. 😸  https://github.com/kedark3/integration_tests/commit/485e3e58d275049fa1c19bd5c40d0cd7f2c8501a#diff-8dc02060f6e8056874c3e03630e11ff3

Reply:
It probably doesn't matter and the code is simple this way.   I only worry that one day it being on may be the default and turning it off would be the wrong thing.

Reply:
Probably. But the primary reason we need to enable the role is "Add Provider" Fixture disables it and @psav had told me it will be best to disable it after the test, as most(almost all, correct me psav) run with "websocket" disabled, except few special cases.

Reply:
The absolute right thing to do is to return it to the state it was in.   Presently this is always disabled.   Someday it may not be.   Still its not a big deal, but its the kind of thing I worry about biting us later.   However in this case there are so many mitigations to it biting us, we may just be fine.

Reply:
**REQUIRED**
I agree with @jamesooden. You need to get the role status before it will be changed and revert it that original state in tear down.

Reply:
So, I totally get what @jamesooden is saying. I did ask @kedark3 to make it turn it back off again, because it was so vital that it be off for testing. That being said, it actually turns out that turning it off takes a while, and rather than waiting for it we have decided to take a different approach and we temporarily just dismiss ALL notifications. So in all actuality we can probably just leave it on now :) Ironically this is how these things sometimes go.

Reply:
Leaving it ON for now. Not disabling anymore.